test/scripts/hexo/post.ts

```diff
@@ -1452,4 +1452,62 @@ describe('Post', () => {
 
     data.content.should.eql('<a href="https://hexo.io/" title="tttitle" target="">foobar</a>');
   });
+
+  // https://github.com/hexojs/hexo/issues/5433
+  it('render() - nunjucks nesting in comments', async () => {
+    const content = [
+      'foo',
+      '<!--',
+      '{% raw %}',
+      'test',
+      '{% endraw %}',
+      '-->',
+      'bar'
+    ].join('\n');
+
+    const data = await post.render('', {
+      content,
+      engine: 'markdown'
+    });
+
+    data.content.should.eql([
+      '<p>foo</p>',
+      '<!--',
+      '{% raw %}',
+      'test',
+      '{% endraw %}',
+      '-->',
+      '<p>bar</p>',
+      ''
+    ].join('\n'));
+  });
+
+  // https://github.com/hexojs/hexo/issues/5433
+  it('render() - code fence nesting in comments', async () => {
+    const code = 'alert("Hello world")';
+    const highlighted = highlight(code);
+    const content = [
+      'foo',
+      '<!--',
+      '```',
+      code,
+      '```',
+      '-->',
+      'bar'
+    ].join('\n');
+
+    const data = await post.render('', {
+      content,
+      engine: 'markdown'
+    });
+
+    data.content.should.eql([
+      '<p>foo</p>',
+      '<!--',
+      `<hexoPostRenderCodeBlock>${highlighted}</hexoPostRenderCodeBlock>`,
```

Comment:
I'm not sure, but this doesn't look right, why does the code in the comments need to be highlighted.

----------

Also, it haven't be restore.

![image](https://github.com/user-attachments/assets/087575e0-9b04-4c82-9648-277531a198de)

Reply:
maybe we can solve it in backtick_code_block.ts

Reply:
You are right. I forgot that the code block was processed before post render